### PR TITLE
Source Marketo - filter 'null' byte(s) in HTTP responses

### PR DESCRIPTION
**Pull request by richa-rochna**
## What
*Source Marketo version 1.2.0 fails with _csv.Error: line contains NUL. Fixing this issue.*

## How
*filtering out 'null' byte(s) in HTTP responses*

## Recommended reading order
1. `spec.json`
2. `source.py`

## 🚨 User Impact 🚨
*No breaking changes. 'null' byte(s) will be filtered out from response.*

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Thanks for the contribution @richa-rochna soon the team will review it.

**Comment by bmcgilli** [comment]
@lazebnyi can you please assign for review? this one is very helpful, thanks!

**Comment by marcosmarxm** [comment]
Thanks for the contribution @richa-rochna and sorry for the lack of updates. I'll review during the week 🙇 !

**Comment by lazebnyi** [comment]
@askarpets please take a look

File: airbyte-integrations/connectors/source-marketo/unit_tests/test_source.py [hunk 1]
```diff
@@ -314,3 +314,45 @@ def test_get_updated_state(config, latest_record, current_state, expected_state)
...
+    test_lines = [
+        "Name,Email\n",
+        "John Doe,john.doe@example.com\n",
+        "Jane Doe,jane.doe@example.com\n"
+    ]
+    expected_records = [
+        {"Name": "John Doe", "Email": "john.doe@example.com"},
+        {"Name": "Jane Doe", "Email": "jane.doe@example.com"}
+    ]
+    records = stream.csv_rows(to_iterable(test_lines))
```
Comment (askarpets) [inline airbyte-integrations/connectors/source-marketo/unit_tests/test_source.py:356]: Please delete `to_iterable` function and just use `records = stream.csv_rows(test_lines)` instead as `test_lines` is already an iterable. Thanks!

File: airbyte-integrations/connectors/source-marketo/unit_tests/test_source.py [hunk 2]
```diff
@@ -314,3 +314,45 @@ def test_get_updated_state(config, latest_record, current_state, expected_state)
...
+        "Hello\x00World\n",
+        "Name,Email\n",
+        "John\x00Doe,john.doe@example.com\n"
+    ]
+    expected_lines = [
+        "HelloWorld\n",
+        "Name,Email\n",
+        "JohnDoe,john.doe@example.com\n"
+    ]
+    filtered_lines = stream.filter_null_bytes(to_iterable(test_lines))
```
Comment (askarpets) [inline airbyte-integrations/connectors/source-marketo/unit_tests/test_source.py:335]: Please delete `to_iterable` function and just use `filtered_lines = stream.filter_null_bytes(test_lines)` instead as `test_lines` is already an iterable. Thanks!

**Review by askarpets — CHANGES_REQUESTED** [review]
@richa-rochna, thank you for your PR! I left some small comments, but overall looks good!

File: airbyte-integrations/connectors/source-marketo/unit_tests/test_source.py (hunk 2 above)
Comment (richa-rochna) [reply airbyte-integrations/connectors/source-marketo/unit_tests/test_source.py]: Sure! 
Done. Please have a look.

File: airbyte-integrations/connectors/source-marketo/unit_tests/test_source.py (hunk 1 above)
Comment (richa-rochna) [reply airbyte-integrations/connectors/source-marketo/unit_tests/test_source.py]: Sure! 
Done. Please have a look.

Comment (askarpets) [reply airbyte-integrations/connectors/source-marketo/unit_tests/test_source.py]: Thank you!

File: airbyte-integrations/connectors/source-marketo/unit_tests/test_source.py (hunk 2 above)
Comment (askarpets) [reply airbyte-integrations/connectors/source-marketo/unit_tests/test_source.py]: Thank you!

**Review by askarpets — APPROVED** [review]
Approving this PR, but please resolve the conflicts.

**Comment by askarpets** [comment]
Hi @marcosmarxm, could you please merge this PR?

**Pull request closed without merging**